src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -248,14 +243,38 @@ internal static void FailFast(string message, Exception? exception, RhFailFastRe
                     outputMessage = message;
                 }
 
+
                 Internal.Console.Error.Write(prefix);
                 if (outputMessage != null)
                     Internal.Console.Error.Write(outputMessage);
                 Internal.Console.Error.Write(Environment.NewLine);
 
-#if FEATURE_DUMP_DEBUGGING
-                GenerateExceptionInformationForDump(exception, IntPtr.Zero);
-#endif
+                if (outputMessage != null)
+                {
+                    // Try to save the exception stack trace in a buffer on the stack.  If the exception is too large, we'll truncate it.
+                    const int MaxStack = 2048;
+                    Span<byte> exceptionStack = stackalloc byte[MaxStack];
+
+                    var utf8Encoded = Encoding.UTF8.GetBytes(outputMessage).AsSpan();
```

Comment:
`Utf8.FromUtf16` lets you do this without allocating the extra array on GC heap.